### PR TITLE
Initial logic check for merging 2 entities

### PR DESCRIPTION
**Pull request by gokullan**
## Change Description
This makes modifications to the `anonymizer_engine.py` file to add functionality to merge entities separated by a whitespace. This current commit does not implement some of the items in the checklist - I will make the next required commits that address these at the earliest.

## Issue reference

This PR fixes issue #1090 

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [ ] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by gokullan** [comment]
@microsoft-github-policy-service agree

**Comment by gokullan** [comment]
I have added a test-case. Do let me know if that is the expected behaviour or if any changes are needed.

**Comment by omri374** [comment]
Thanks @gokullan! We will review this PR in the next few days.

File: presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py
Comment (omri374) [inline presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py:82]: Can we move this logic into a function? to make the anonymization pipeline more readable?

File: presidio-anonymizer/tests/test_anonymizer_engine.py
Comment (omri374) [inline presidio-anonymizer/tests/test_anonymizer_engine.py:160]: Could you please add a few more test cases:
- For multiple spaces
- For two entities that are not from the same type but have a whitespace between them
- For two entities of the same type with a character different from whitespace between them

Consider using `@pytest.mark.parametrize`

**Review by omri374 — COMMENTED** [review]
Thank you @gokullan! I've added a few minor suggestions, but this looks great.

**Comment by gokullan** [comment]
Sure, will make the requested changes.

File: presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py
Comment (gokullan) [reply presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py]: I have moved the logic to a function named `_merge_entities_with_whitespace_between`

File: presidio-anonymizer/tests/test_anonymizer_engine.py
Comment (gokullan) [reply presidio-anonymizer/tests/test_anonymizer_engine.py]: I have added the test cases, but I'm not sure if the way I have currently written them is very elegant. Do let me know if I need to modify. (Commit [8620534](https://github.com/microsoft/presidio/pull/1092/commits/86205342be1e9ec909a78e5bc2e9033cf48c0e4b))

Comment (omri374) [reply presidio-anonymizer/tests/test_anonymizer_engine.py]: Looks good, thanks!

**Comment by omri374** [comment]
waiting for a second approval to approve @navalev @SharonHart @shiranr

**Review by SharonHart — APPROVED** [review]

**Pull request merged**
